File: leetcode/3142-check-if-grid-satisfies-conditions/satisfiesConditions.ts
function satisfiesConditions(grid: number[][]): boolean {
  let rowLength = grid.length;
  let colLength = grid[0].length;

  for (let row = 0; row < rowLength; row++) {
    for (let col = 0; col < colLength; col++) {
      const cell = grid[row][col];

      if (row < rowLength - 1) {
        if (cell != grid[row + 1][col]) {
          return false;
        }
      }

      if (col < colLength - 1) {
        if (cell == grid[row][col - 1]) {
          return false;
        }
      }
    }
  }

  return true;
}

export default satisfiesConditions;

File: leetcode/3142-check-if-grid-satisfies-conditions/satisfiesConditions.rs
use crate::Solution;

impl Solution {
    pub fn satisfies_conditions(grid: Vec<Vec<i32>>) -> bool {
        let row_length = grid.len();
        let col_length = grid[0].len();

        for row in 0..row_length {
            for col in 0..col_length {
                let cell = grid[row][col];

                if row < row_length - 1 {
                    if cell != grid[row + 1][col] {
                        return false;
                    }
                }

                if col < col_length - 1 {
                    if cell == grid[row][col + 1] {
                        return false;
                    }
                }
            }
        }

        return true;
    }
}
